internal/attestation/maa_test.go

```diff
@@ -0,0 +1,255 @@
+				Expiry:    jwt.NewNumericDate(time.Now().Add(time.Hour)),
+				IssuedAt:  jwt.NewNumericDate(time.Now()),
+				Issuer:    "wrong issuer",
+				ID:        "222000",
+				NotBefore: jwt.NewNumericDate(time.Now()),
+			},
+			key, true,
+		},
+		{
+			"expired",
+			jwt.Claims{
+				Expiry:    jwt.NewNumericDate(time.Now().Add(-1 * time.Hour)),
+				IssuedAt:  jwt.NewNumericDate(time.Now().Add(-2 * time.Hour)),
+				Issuer:    attestationProvider.URL,
+				ID:        "222000",
+				NotBefore: jwt.NewNumericDate(time.Now().Add(-2 * time.Hour)),
+			},
+			key, true,
+		},
+		{
+			"before nbf",
+			jwt.Claims{
+				Expiry:    jwt.NewNumericDate(time.Now().Add(2 * time.Hour)),
+				IssuedAt:  jwt.NewNumericDate(time.Now().Add(-2 * time.Hour)),
+				Issuer:    attestationProvider.URL,
+				ID:        "222000",
+				NotBefore: jwt.NewNumericDate(time.Now().Add(time.Hour)),
+			},
+			key, true,
+		},
+		{
+			"issued in future",
+			jwt.Claims{
+				Expiry:    jwt.NewNumericDate(time.Now().Add(4 * time.Hour)),
+				IssuedAt:  jwt.NewNumericDate(time.Now().Add(2 * time.Hour)),
+				Issuer:    attestationProvider.URL,
+				ID:        "222000",
+				NotBefore: jwt.NewNumericDate(time.Now().Add(2 * time.Hour)),
+			},
+			key, true,
+		},
+		{
+			"wrong signature",
+			jwt.Claims{
+				Expiry:    jwt.NewNumericDate(time.Now().Add(4 * time.Hour)),
+				IssuedAt:  jwt.NewNumericDate(time.Now().Add(2 * time.Hour)),
+				Issuer:    attestationProvider.URL,
+				ID:        "222000",
+				NotBefore: jwt.NewNumericDate(time.Now().Add(2 * time.Hour)),
+			},
+			evilKey, true,
+		},
+	}
+	for _, test := range tests {
+		t.Logf("Subtest: %v", test.Name)
+		//
+		// Create token.
+		//
+		publicClaims := test.PublicClaims
+		privateClaims := privateClaims{
+			Data:            base64.RawURLEncoding.EncodeToString(data),
+			Debug:           true,
+			UniqueID:        string(uniqueID),
+			SignerID:        string(signerID),
+			ProductID:       123,
+			SecurityVersion: 321,
+		}
+		sig, err := jose.NewSigner(jose.SigningKey{Algorithm: jose.RS256, Key: test.Key}, (&jose.SignerOptions{}).WithType("JWT").WithHeader("kid", "aaa"))
+		require.NoError(err)
+		rawToken, err := jwt.Signed(sig).Claims(publicClaims).Claims(privateClaims).CompactSerialize()
+		require.NoError(err)
+		//
+		// Verify token and check report.
+		//
+		report, err := VerifyAzureAttestationToken(rawToken, attestationProvider.URL)
+		if test.ExpectErr {
+			require.Error(err)
+			continue
+		} else {
+			require.NoError(err)
+		}
+		require.Equal(data, report.Data)
+		require.Equal(privateClaims.SecurityVersion, report.SecurityVersion)
+		require.Equal(privateClaims.Debug, report.Debug)
+		require.Equal(uniqueID, report.UniqueID)
+		require.Equal(signerID, report.SignerID)
+		productID := make([]byte, 16)
+		binary.LittleEndian.PutUint16(productID, uint16(privateClaims.ProductID))
+		require.Equal(productID, report.ProductID)
+	}
+}
+
+func TestSharedProviderKeyParsing(t *testing.T) {
```

Comment:
```suggestion
func DisabledTestSharedProviderKeyParsing(t *testing.T) {
```
Unit tests should be self-contained. Either disable this test or hardcode the response.